### ISSUE TITLE
[Supabase][cookies] Cannot read properties of undefined (reading 'name')

### ISSUE THREAD

**Issue by LeonardoBevilacqua** [vercel/next.js#58026]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Thu, 02 Nov 2023 12:57:06 +0000
Binaries:
  Node: 20.9.0
  npm: 10.2.3
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.2-canary.11
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which example does this report relate to?

with-supabase

### What browser are you using? (if relevant)

Mozilla Firefox for Arch Linux 119.0

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

The default page is not loading. when running in develop mode the error is displayed.

```
Unhandled Runtime Error

Error: Cannot read properties of undefined (reading 'name')
utils/supabase/server.ts (11:29) @ get

     9 | cookies: {
   10 |   get(name: string) {
> 11 |     return cookieStore.get(name)?.value
         |                                      ^
   12 |   },
   13 |   set(name: string, value: string, options: CookieOptions) {
   14 |     try {
```

### Expected Behavior

Expected the page to load without issues

### To Reproduce

run `npx create-next-app -e with-supabase`, set up the .env.local following the README instructions, then try to run with `npm run dev`

**Comment by coleerikson** [comment]
I'm also running into this issue. When you update `process.env.NEXT_PUBLIC_SUPABASE_URL` the error occurs - any non-url string gives it the appearance the app is working.

**Comment by arinaldi** [comment]
I bypassed this error by making the following update in the `utils/supabase/middleware` and `utils/supabase/server` files:

```ts
get(name: string) {
  if (!name) return;
  return request.cookies.get(name)?.value;
},
```

After signing in with email and password, the session is populated when accessing it from `utils/supabase/client`.

However, the session is `null` when accessing it from `utils/supabase/middleware` or `utils/supabase/server`.

My theory is those two files are calling their `get` method without a valid argument, and the session cookie cannot be retrieved.

**Comment by LeonardoBevilacqua** [comment]
The workaround above works but still, when trying to sign in, the access token is set with the name 'undefined'
![image](https://github.com/vercel/next.js/assets/24481966/e0aef361-bad5-497d-b1fa-0ccae3958be8)

**Comment by jonvet** [comment]
works for me if I downgrade to `"@supabase/ssr": "^0.0.6"`

**Comment by j13n** [comment]
Same issue here but with Sveltekit on `"@supabase/ssr": "0.0.8"`

![image](https://github.com/vercel/next.js/assets/5645903/a8754f59-3362-4570-9d4e-5cd4e5957cf0)

**Comment by LeonardoBevilacqua** [comment]
[@supabase/ssr NextJs 14 middleware broke in 0.0.8](https://github.com/supabase/auth-helpers/issues/676)

**Comment by coleerikson** [comment]
Version `0.0.6` seems to fix it. Thank you everyone. I thought I'd have to wait until Monday to hear a response. 🙏

**Comment by arinaldi** [comment]
The issue is fixed in `@supabase/ssr` version `0.0.9`.

**Comment by timneutkens** [comment]
Closing as I've checked with the Supabase team and they said it's fixed 👍
